pandas/core/groupby/generic.py

```diff
@@ -1221,10 +1221,18 @@ def _apply_to_column_groupbys(self, func):
         return func(self)
 
     def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None):
-        """Calculate percent change of each value to previous entry in group"""
+        """Calcuate pct_change of each value to previous entry in group"""
+        if freq:
+            return self.apply(lambda x: x.pct_change(periods=periods,
+                                                     fill_method=fill_method,
+                                                     limit=limit, freq=freq))
         filled = getattr(self, fill_method)(limit=limit)
-        shifted = filled.shift(periods=periods, freq=freq)
+        fill_grp = filled.groupby(self.grouper.labels)
+        shifted = fill_grp.shift(periods=periods, freq=freq)
 
+        # TODO: Remove this conditional when #23918 is fixed
```

Comment:
I don't think you need this and the condition on `freq` right? OK to just have the former but move this comment up

Reply:
Right, good point. 
Although fixing #23918 won't necessarily solve the problem here, right? As even if it does not return a multi-index, it still won't handle division as you pointed up to. 

For now I'll just stick to the relegation to the anonymous function, and remove the check for the multi-index.

Reply:
(which is what I think you were suggesting, but correct me if I'm wrong)

Reply:
You've got it